packages/kbn-config-mocks/src/deprecations.mock.ts

```diff
@@ -7,7 +7,9 @@
  */
 
 import type { DocLinks } from '@kbn/doc-links';
-import type { ConfigDeprecationContext } from './types';
+import type { ConfigDeprecationContext } from '@kbn/config';
+
+export type ConfigDeprecationContextMock = ConfigDeprecationContext;
```

Comment:
Exporting dedicated mock type, even if for now it's the exact same as the concrete implementation